### DISCUSSION TITLE
Change URL without reloading app

### DISCUSSION THREAD

**Discussion by kostazol** [vercel/next.js#25061] [Help]
### Describe the feature you'd like to request

Change URL without reloading app

### Describe the solution you'd like

Before Next.js i used angularjs. And I like one feature. I want to change URL link in my browser without reloading page. I used ui-router library. There is react example:
https://ui-router.github.io/sample-app-react/#/welcome
and i can edit link when I opened page and get this link https://ui-router.github.io/sample-app-react/#/home without reloading app but with changing page. Can I config Next.js router the same? 

### Describe alternatives you've considered

I tried https://nextjs.org/docs/routing/shallow-routing but it can't solved this problem.

**Comment by bobarros** [comment]
Did you try this?

```jsx
window.history.replaceState(null, "", "/your/new/path")
```

**Reply by kostazol** [reply]
If I change URL from my app, for example by button click link "/home" it works perfect and it doesn't reload app, but if I manual change url app will be  reloaded.
